src/burn/engine/plan.cpp

```diff
@@ -1033,6 +1037,9 @@ extern "C" HRESULT PlanCachePackage(
 
     if (fBARequestedCache || NeedsCache(pPlan, pPackage))
     {
+        // The behavior for cache only mode is to do nothing on rollback, e.g. for subsequent install on demand scenarios.
```

Comment:
Just about to update the PR. I'd recommend, if you want to discuss further, we bring it up at the next triage and file it as a feature bug. This needs to get checked in and merging grows difficult.